runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WindowIntoTranslation.java

```diff
@@ -105,11 +106,10 @@ public static WindowIntoPayload getWindowIntoPayload(AppliedPTransform<?, ?, ?>
         getWindowIntoPayload(application).getWindowFn());
   }
 
-  /**
-   * A {@link TransformPayloadTranslator} for {@link Window}.
-   */
+  /** A {@link TransformPayloadTranslator} for {@link Window}. */
   public static class WindowIntoPayloadTranslator
-      implements PTransformTranslation.TransformPayloadTranslator<Window.Assign<?>> {
+      extends PTransformTranslation.TransformPayloadTranslator.WithDefaultRehydration<
+          Window.Assign<?>> {
```

Comment:
<!--new_thread; commit:042367c660b607e55c9faafa4d3d8a8685547151; resolved:0-->
As an aside, I'm moderately surprised at this, but it's only because I forgot we carry the WindowFn by value in both the WindowingStrategy and the WindowIntoPayload